connector/connector_github.go

```diff
@@ -0,0 +1,248 @@
+package connector
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/dex/pkg/log"
+	"github.com/coreos/go-oidc/oauth2"
+	"github.com/coreos/go-oidc/oidc"
+)
+
+const (
+	GitHubConnectorType  = "github"
+	gitHubAuthorizeURL   = "https://github.com/login/oauth/authorize"
+	gitHubAccessTokenURL = "https://github.com/login/oauth/access_token"
+	gitHubAPIUserURL     = "https://api.github.com/user"
+)
+
+func init() {
+	RegisterConnectorConfigType(GitHubConnectorType, func() ConnectorConfig { return &GitHubConnectorConfig{} })
+}
+
+type GitHubConnectorConfig struct {
+	ID           string   `json:"id"`
+	ClientID     string   `json:"clientID"`
+	ClientSecret string   `json:"clientSecret"`
+	Scopes       []string `json:"scopes"`
+}
+
+func (cfg *GitHubConnectorConfig) ConnectorID() string {
+	return cfg.ID
+}
+
+func (cfg *GitHubConnectorConfig) ConnectorType() string {
+	return GitHubConnectorType
+}
+
+type GitHubConnector struct {
+	id           string
+	cbURL        url.URL
+	loginFunc    oidc.LoginFunc
+	clientID     string
+	clientSecret string
+	scopes       string // comma separated list of scopes
+}
+
+func (cfg *GitHubConnectorConfig) Connector(ns url.URL, lf oidc.LoginFunc, tpls *template.Template) (Connector, error) {
+	ns.Path = path.Join(ns.Path, httpPathCallback)
+	return &GitHubConnector{
+		id:           cfg.ID,
+		cbURL:        ns,
+		loginFunc:    lf,
+		clientID:     cfg.ClientID,
+		clientSecret: cfg.ClientSecret,
+		scopes:       strings.Join(cfg.Scopes, ","),
+	}, nil
+}
+
+func (c *GitHubConnector) ID() string {
+	return c.id
+}
+
+func (c *GitHubConnector) Healthy() error {
```

Comment:
Yeah I noticed that when attempting to write the function. Seems like a better idea to track rate limiting, since this could actually prevent the connector from doing its job.